### PR TITLE
Follow up changes on secrets patch

### PR DESCRIPTION
**Pull request by umohnani8**
Deleted mounts.conf file and moved the secrets mount paths
to a list (default-mounts) in crio.conf

Signed-off-by: umohnani8 <umohnani@redhat.com>

### CODE REVIEW

**Comment by umohnani8** [comment]
@rhatdan @mrunalp @runcom PTAL

**Comment by umohnani8** [comment]
follow up changes from https://github.com/kubernetes-incubator/cri-o/pull/1003

File: cmd/crio/config.go [hunk 1]
```diff
@@ -108,8 +108,9 @@ cgroup_manager = "{{ .CgroupManager }}"
 # hooks_dir_path is the oci hooks directory for automatically executed hooks
 hooks_dir_path = "{{ .HooksDirPath }}"
 
-# default_mounts_path is the secrets mounts file path
-default_mounts_path = "{{ .DefaultMountsPath }}"
+# default_mountsis the secrets mounts file path
```
Comment (TomSweeneyRedHat) [inline cmd/crio/config.go]: nit "default_mounts is"

Comment (umohnani8) [reply cmd/crio/config.go]: fixed

File: libkpod/config.go [hunk 2]
```diff
@@ -295,7 +292,6 @@ func DefaultConfig() *Config {
 			ContainerExitsDir: containerExitsDir,
 			HooksDirPath:      DefaultHooksDirPath,
 			LogSizeMax:        DefaultLogSizeMax,
```
Comment (TomSweeneyRedHat) [inline libkpod/config.go]: Do we need DefaultMounts in here?

Comment (umohnani8) [reply libkpod/config.go]: Nope, it will be set by the user crio.conf when they want to use it.

File: cmd/crio/main.go [hunk 3]
```diff
@@ -325,8 +325,8 @@ func main() {
 			Value:  libkpod.DefaultHooksDirPath,
 			Hidden: true,
 		},
-		cli.StringFlag{
-			Name:   "default-mounts-path",
+		cli.StringSliceFlag{
+			Name:   "default-mounts",
 			Usage:  "set the default mounts file path",
```
Comment (mrunalp) [inline cmd/crio/main.go]: This isn't a file path anymore but actual mounts host:container.

Comment (umohnani8) [reply cmd/crio/main.go]: right, fixed.

File: libkpod/config.go (hunk 2 above)
Comment (TomSweeneyRedHat) [reply libkpod/config.go]: Ok great, thx for the 411

File: test/helpers.bash [hunk 4]
```diff
@@ -69,12 +69,14 @@ HOOKSDIR=$TESTDIR/hooks
...
-touch ${DEFAULT_MOUNTS_FILE}
-DEFAULT_MOUNTS_OPTS="--default-mounts-path=$DEFAULT_MOUNTS_FILE"
+# Setup default secrets mounts
+MOUNT_PATH="$TESTDIR/secrets"
+mkdir ${MOUNT_PATH}
+MOUNT_FILE="${MOUNT_PATH}/test.txt"
+touch ${MOUNT_FILE}
+echo "Testing secrets mounts!" > ${MOUNT_FILE}
+
+DEFAULT_MOUNTS_OPTS="--default-mounts=${MOUNT_PATH}:/container/path1"
```
Comment (rhatdan) [inline test/helpers.bash]: Don't you need to modify the crio config line in helpers.bash?

File: cmd/crio/main.go [hunk 5]
```diff
@@ -325,9 +325,9 @@ func main() {
 			Value:  libkpod.DefaultHooksDirPath,
 			Hidden: true,
 		},
-		cli.StringFlag{
-			Name:   "default-mounts-path",
-			Usage:  "set the default mounts file path",
+		cli.StringSliceFlag{
+			Name:   "default-mounts",
+			Usage:  "set the default mounts paths in the form host:container",
```
Comment (rhatdan) [inline cmd/crio/main.go]: add a default mount path in the form host:container

**Comment by rhatdan** [comment]
docs/crio.8.md and docs/crio.5.md need to be updated.

**Comment by umohnani8** [comment]
> docs/crio.8.md and docs/crio.5.md need to be updated.
@rhatdan `default-mounts` is a hidden flag. Do we want that to be visible now? The docs don't have anything on `hooks-dir` since it is hidden, so what exatlty should I update in the docs?

File: test/helpers.bash (hunk 4 above)
Comment (umohnani8) [reply test/helpers.bash]: I don't think so, that just tells it where the conf file is in the test right? I modified the line where crio is being built, so it is built with the `default-mounts` flag for testing purposes.

File: cmd/crio/main.go (hunk 5 above)
Comment (umohnani8) [reply cmd/crio/main.go]: fixed

**Comment by rhatdan** [comment]
Ok don't document the CLI, but you need to document the configuration.

**Comment by umohnani8** [comment]
@rhatdan updated configuration doc

File: docs/crio.conf.5.md [hunk 6]
```diff
@@ -105,6 +105,10 @@ Example:
 **no_pivot**=*true*|*false*
   Instructs the runtime to not use pivot_root, but instead use MS_MOVE
 
+**default_mounts**=[]
+  Default mounts list for the mount paths, in the form host:container, to be mounted
```
Comment (rhatdan) [inline docs/crio.conf.5.md]: List of mount points, in the form host:container, to be mounted in every container

Comment (umohnani8) [reply docs/crio.conf.5.md]: fixed

**Comment by rhatdan** [comment]
LGTM
@mrunalp PTAL

**Comment by mrunalp** [comment]
Looks fine.

**Comment by mrunalp** [comment]
LGTM

**Pull request merged**